src/transaction/log_manager.c

```diff
@@ -10214,6 +10219,482 @@ logtb_tran_update_stats_online_index_rb (THREAD_ENTRY * thread_p, void *data, vo
+		  *time = archive_start_time;
+		  is_found = true;
+		  error = ER_CDC_LSA_NOT_FOUND;	/*should be replaced */
+		}
+	      else
+		{
+		  if ((error = get_lsa_with_start_point (thread_p, time, &ret_lsa)) != NO_ERROR)
+		    {
+		      is_found = true;
+		      LSA_COPY (start_lsa, &ret_lsa);
+		    }
+		}
+	    }
+	  else
+	    {
+	      /* num_arvs == 0, and active_start_time > input time 
+	       * returns oldest LSA in active log volume */
+	      *time = active_start_time;
+	      LSA_COPY (start_lsa, &ret_lsa);
+	      is_found = true;
+	    }
+	}
+    }
+
+  if (is_found)
+    {
+      LSA_COPY (&log_Reader_info.start_lsa, start_lsa);
+      /*pthread_cond_signal () to cdc_log_producer and queue re-initialization */
+    }
+
+  return error;
+}
+
+/*
+ * arv_num (in) : archive log volume number to traverse. If it is -1, then traverse active log volume. 
+ * ret_lsa (out) : lsa of the first log which contains time info 
+ * time (out) : time of the first log which contains time info  
+ */
+
+static int
+get_start_point_from_file (THREAD_ENTRY * thread_p, int arv_num, LOG_LSA * ret_lsa, time_t * time)
+{
+  char arv_name[PATH_MAX];
+  LOG_ARV_HEADER *arv_hdr;
+  char hdr_pgbuf[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT], *aligned_hdr_pgbuf;
+  char log_pgbuf[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT], *aligned_log_pgbuf;
+
+  LOG_PAGE *hdr_pgptr;
+  LOG_PAGE *log_pgptr;
+  LOG_PHY_PAGEID phy_pageid = NULL_PAGEID;
+  int vdes;
+
+  LOG_LSA process_lsa;
+  LOG_LSA forw_lsa;
+
+  LOG_RECORD_HEADER *log_rec_header;
+  LOG_REC_DONETIME *donetime;
+  LOG_REC_HA_SERVER_STATE *dummy;
+
+  aligned_log_pgbuf = PTR_ALIGN (log_pgbuf, MAX_ALIGNMENT);
+  log_pgptr = (LOG_PAGE *) aligned_log_pgbuf;
+  LOG_CS_ENTER_READ_MODE (thread_p);
+
+  if (arv_num == -1)
+    {
+      process_lsa.pageid = log_Gl.hdr.fpageid;
+      process_lsa.offset = 0;
+    }
+  else
+    {
+      LOG_ARCHIVE_CS_ENTER (thread_p);
+      aligned_hdr_pgbuf = PTR_ALIGN (hdr_pgbuf, MAX_ALIGNMENT);
+
+      hdr_pgptr = (LOG_PAGE *) aligned_hdr_pgbuf;
+
+      fileio_make_log_archive_name (arv_name, log_Archive_path, log_Prefix, arv_num);
+
+      if (fileio_is_volume_exist (arv_name) == true)
+	{
+	  vdes = fileio_mount (thread_p, log_Db_fullname, arv_name, LOG_DBLOG_ARCHIVE_VOLID, false, false);
+	  if (vdes != NULL_VOLDES)
+	    {
+	      if (fileio_read (thread_p, vdes, hdr_pgptr, 0, LOG_PAGESIZE) == NULL)
+		{
+		  fileio_dismount (thread_p, vdes);
+
+		  er_set (ER_FATAL_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_READ, 3, 0LL, 0LL, arv_name);
+
+		  LOG_ARCHIVE_CS_EXIT (thread_p);
+
+		  return ER_FAILED;
+		}
+
+	      arv_hdr = (LOG_ARV_HEADER *) hdr_pgptr->area;
+	      process_lsa.pageid = arv_hdr->fpageid;
+	      process_lsa.offset = 0;
+
+	      fileio_dismount (thread_p, vdes);
+	      LOG_ARCHIVE_CS_EXIT (thread_p);
+	    }
+	}
+    }
+
+  LOG_CS_EXIT (thread_p);
+
+  if (logpb_fetch_page (thread_p, &process_lsa, LOG_CS_SAFE_READER, log_pgptr) != NO_ERROR)
+    {
+      return ER_FAILED;
```

Comment:
Why ER_FAILED should be returned instead of the resulting error from `logpb_fetch_page`?